### PR TITLE
Change engine -> {msg,chain} service comms to be synchronous 

### LINKED ISSUE #720: Audit use of channels in the engine
The chain service and the messaging service currently communicate _in both directions_ with `go chans`. 

It might make sense to use synchronous function calls instead when the engine needs to send information out to either service. 

We could keep channels for the other direction, which makes more sense.

### PR DESCRIPTION
**Pull request by geoknee**
Closes #720 . 

Possibly relevant to #704 , #700 .

Pros:
- easier to debug this code because fewer go routines
- easier to reason about control flow of the engine
- lower probability to have a race condition?

Cons
- _might_ have an adverse effect on performance (anecdotal evidence)
- may increase probability for a dead lock?

### CODE REVIEW

File: client/engine/engine.go [hunk 1]
```diff
@@ -325,11 +324,11 @@ func (e *Engine) handleAPIEvent(apiEvent APIEvent) (ObjectiveChangeEvent, error)
 func (e *Engine) executeSideEffects(sideEffects protocols.SideEffects) {
 	for _, message := range sideEffects.MessagesToSend {
 		e.logger.Printf("Sending message %+v", protocols.SummarizeMessage(message))
-		e.toMsg <- message
+		e.msg.Send(message)
 	}
 	for _, tx := range sideEffects.TransactionsToSubmit {
 		e.logger.Printf("Sending chain transaction for channel %s", tx.ChannelId)
-		e.toChain <- tx
+		e.chain.Send(tx)
```
Comment (geoknee) [inline client/engine/engine.go:331]: This is the nub of the change.

File: client/engine/chainservice/chainservice.go [hunk 2]
```diff
@@ -44,6 +44,8 @@ type ChainEventHandler interface {
 }
 
 type ChainService interface {
+	// Out returns a chan for receiving events from the chain service
 	Out() <-chan Event
-	In() chan<- protocols.ChainTransaction
+	// Send is for sending transactions with the message service
```
Comment (NiloCK) [inline client/engine/chainservice/chainservice.go:49]:
```suggestion
	// Send is for sending transactions with the chain service
```

File: client/engine/chainservice/simplechainservice.go [hunk 3]
```diff
@@ -34,16 +31,9 @@ func (mcs simpleChainService) Out() <-chan Event {
...
-}
-
-// forwardTransactions pipes transactions to the MockChain
-func (mcs simpleChainService) forwardTransactions() {
-	for tx := range mcs.in {
-		mcs.chain.In() <- tx
-	}
+// Send pipes transactions to the MockChain
+func (mcs simpleChainService) Send(tx protocols.ChainTransaction) {
+	mcs.chain.In() <- tx // TODO block until this has been successful / convert chain.In to a sync call
```
Comment (NiloCK) [inline client/engine/chainservice/simplechainservice.go:36]: I believe this does block until successful. 

https://go.dev/tour/concurrency/2:
> By default, sends and receives block until the other side is ready.

ie, this code does not advance until `In()` is ready to consume `tx`.

```suggestion
	mcs.chain.In() <- tx // TODO convert chain.In to a sync call
```

**Review by NiloCK — APPROVED** [review]
This also LGTM.

I am seeing a promising reduction in test failures locally with this branch. :crossed_fingers: 

(It is disturbingly easy to autocomplete a middle finger when attempting those crossed fingers.)

**Pull request merged**